pkg/auth/auth.go

```diff
@@ -154,6 +167,15 @@ func getUserAndPass(opts *LoginOptions, password, userFromAuthFile string) (stri
 // Logout removes the authentication of server from authfile
 // removes all authtication if specifies all in the options
 func Logout(systemContext *types.SystemContext, opts *LogoutOptions, server string) error {
+	if server == "" && !opts.All {
+		registriesFromFile, err := registries.GetRegistries()
+		if err != nil || len(registriesFromFile) == 0 {
+			return errors.Errorf("no registries found in registries.conf, a registry must be provided")
+		}
+
+		server = registriesFromFile[0]
+		logrus.Debugf("registry not specified, default to the first registry %q from registries.conf", server)
+	}
 	if server != "" {
 		server = getRegistryName(server)
```

Comment:
This should not be called on the `registriesFromFile[0]` value obtained above, to be consistent with `Login`.

Reply:
This is still outstanding.

Reply:
> This should not be called on the `registriesFromFile[0]` value obtained above, to be consistent with `Login`.

what do you mean, it should not from registry.conf?

Reply:
Consider `Logout(server="")`: `defaultRegistryWhenUnspecified` will return a value, this `if server != ""` will be `true`, and the return value will be processed through `getRegistryName`. That’s both unnecessary and inconsistent with the login path.